src/frontend/components/Chat/SharedChatComponents/ChatInputBar/index.tsx

```diff
@@ -15,78 +12,62 @@ const messages = defineMessages({
+  },
 });
 
-const SendButtonStyled = styled(Button)`
-  width: 40px;
-  height: 40px;
-  border: 1px solid;
-  border-color: ${normalizeColor('blue-chat', theme)};
-  border-radius: 6px;
-  margin: 5px;
-`;
+interface ChatInputBarProps {
+  isActive: boolean;
+  setOverlay: React.Dispatch<React.SetStateAction<boolean>>;
+}
 
-export const ChatInputBar = () => {
-  const [chatMessage, setChatMessage] = useState('');
+export const ChatInputBar = ({ isActive, setOverlay }: ChatInputBarProps) => {
   const intl = useIntl();
 
-  const handleSendMessage = async () => {
-    if (chatMessage.length !== 0) {
-      await converse.sendMessage(chatMessage);
-      setChatMessage('');
-    }
+  const handleClickOnJoinButton = () => {
+    setOverlay(true);
   };
 
-  const handleKeypress = (event: React.KeyboardEvent<HTMLInputElement>) => {
-    if (event.code === 'Enter') {
-      handleSendMessage();
-    }
+  const processChatMessage = (chatMsg: string): number => {
+    converse.sendMessage(chatMsg);
+    return 1;
   };
 
   return (
-    <Box
-      direction="row"
-      align="center"
-      border={{
-        size: 'xsmall',
-        color: 'blue-chat',
-      }}
-      round="6px"
-      margin="10px"
-      height="50px"
-    >
-      <TextInput
-        placeholder={
-          <Text color="blue-chat" size="16px">
-            {intl.formatMessage(messages.inputBarPlaceholder)}
-          </Text>
-        }
-        value={chatMessage}
-        onChange={(event) => setChatMessage(event.target.value)}
-        plain={true}
-        focusIndicator={false}
-        onKeyPress={(keyPressEvent) => handleKeypress(keyPressEvent)}
-        size="16px"
-      />
-      <SendButtonStyled
-        onClick={() => handleSendMessage()}
-        icon={
-          <Box
-            margin={{
-              vertical: '2px',
-              horizontal: '-7px',
-            }}
-          >
-            <SendButtonSVG
-              iconColor={'blue-chat'}
-              containerStyle={{
-                width: '25px',
-                height: '22px',
-              }}
-            />
-          </Box>
-        }
-      />
+    <Box margin="10px">
+      {isActive ? (
+        <InputBar
+          handleUserInput={processChatMessage}
+          isChatInput={true}
+          placeholderText={intl.formatMessage(messages.inputBarPlaceholder)}
+        />
+      ) : (
+        <Box
+          align="center"
+          border={{
+            color: 'blue-chat',
+            size: 'xsmall',
+          }}
+          height="50px"
+          round="6px"
+        >
+          <Button fill={true} onClick={handleClickOnJoinButton} plain={true}>
+            <Box>
+              <Text
+                color="blue-chat"
+                size="17px"
+                textAlign="center"
+                truncate={true}
+              >
+                {intl.formatMessage(messages.joinChatButton)}
+              </Text>
+            </Box>
+          </Button>
```

Comment:
I'm not using the label here, because there is not only text but also layout components and I found the code more readable this way. Indeed, the button encapsulates the Box and the Text components so to me, it feels like it's more intuitive to pass these components as children instead of props.